src/azure-cli-core/setup.py

```diff
@@ -47,6 +47,7 @@
     'argcomplete~=1.8',
     'azure-cli-telemetry==1.0.6.*',
     'colorama~=0.4.1',
+    'cryptography>=3.2,<3.4',
```

Comment:
`adal` relies on `cryptography`, so moved it to `azure-cli-core`.

`3.4+` versions rely on rust and cause install issues on some platforms. Let's set the upper bound for now.

Reply:
> `3.4+` versions rely on rust and cause install issues on some platforms. Let's set the upper bound for now.

The original discussion: https://github.com/pyca/cryptography/issues/5771

Reply:
`azure-devops` (#16858) and `azure-cli-ml` (#16880) shouldn't rely on `adal` or `cryptography` directly. They should use the those provided by `azure-cli-core`.

Reply:
Good point. I will submit issues to `azure-devops`. Machine lerning team is working on a new extension to replace the old one, I will request a cleaning of their dependnecies.